kernel/comps/time/src/drivers/cmos.rs

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use core::sync::atomic::{AtomicU8, Ordering::Relaxed};
+
+use ostd::arch::x86::device::cmos::{century_register, CMOS_ADDRESS, CMOS_DATA};
+
+use crate::{SystemTime, drivers::Driver};
+
+static CENTURY_REGISTER: AtomicU8 = AtomicU8::new(0);
+
+fn get_cmos(reg: u8) -> u8 {
+    CMOS_ADDRESS.write(reg);
+    CMOS_DATA.read()
+}
+
+fn is_updating() -> bool {
+    CMOS_ADDRESS.write(0x0A);
+    CMOS_DATA.read() & 0x80 != 0
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+struct CmosData {
+    century: u8,
+    year: u16,
+    month: u8,
+    day: u8,
+    hour: u8,
+    minute: u8,
+    second: u8,
+}
+
+impl CmosData {
+    fn from_rtc_raw(century_register: u8) -> Self {
+        while is_updating() {}
+
+        let second = get_cmos(0x00);
+        let minute = get_cmos(0x02);
+        let hour = get_cmos(0x04);
+        let day = get_cmos(0x07);
+        let month = get_cmos(0x08);
+        let year = get_cmos(0x09) as u16;
+
+        let century = if century_register != 0 {
+            get_cmos(century_register)
+        } else {
+            0
+        };
+
+        CmosData {
+            century,
+            year,
+            month,
+            day,
+            hour,
+            minute,
+            second,
+        }
+    }
+
+    /// convert BCD to binary values
```

Comment:
-> `Converts BCD to binary values.`
Documents should begin with the third person singular and should begin with a capital letter. These are actually issues that existed before, but the code changes in this PR can be corrected if it's convenient.